Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerSuggestionsViewController.swift

```diff
@@ -6,26 +6,28 @@ import StreamChat
 import UIKit
 
 public protocol SuggestionsViewControllerPresenter: class {
-    func showSuggestionsViewController(
-        with state: SuggestionsViewControllerState,
-        onSelectItem: ((Int) -> Void)
-    )
+    func showSuggestionsViewController(for kind: SuggestionKind, onSelectItem: ((Int) -> Void))
     func dismissSuggestionsViewController()
 
     var isSuggestionControllerPresented: Bool { get }
 }
 
-public enum SuggestionsViewControllerState {
-    case commands([Command])
-    case mentions([String])
+public enum SuggestionKind {
+    case command(hints: [Command])
+    case mention
 }
 
 open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>: ViewController,
     UIConfigProvider,
-    UICollectionViewDelegate,
-    UICollectionViewDataSource {
+    UICollectionViewDelegate {
     // MARK: - Property
 
+    public var dataSource: UICollectionViewDataSource? {
+        didSet {
+            collectionView.dataSource = dataSource
+        }
+    }
+
```

Comment:
In this context, the data source should be treated as content. You shouldn't access the collection view before the lifetime methods are colled. So this should rather be `updateContentIfNeeded` and you should assign the data source in `updateContent` (like to already do)

Reply:
Thank you, I fixed this.